src/move/move.ts

```diff
@@ -0,0 +1,23 @@
+const ALLOWED_FIRST_MOVES = [
+  [false, true, true, false],
+  [true, false, false, false, true],
+  [true, false, false, false, false, true],
+  [false, false, false, false, false, false, false],
+  [true, false, false, false, false, true],
+  [true, false, false, false, true],
+  [false, true, true, false],
+];
+
+export interface Action {
+  x: number;
+  y: number;
+  isFirstMove: boolean;
+}
+
+export const checkIfMoveIsAllowed = (action: Action): boolean => {
```

Comment:
You should find another name for this method. To me, `checkIfMoveIsAllowed` would imply that this fn is capable of checking any move, while it is actually only responsible of checking the first move.

Unless you planned to add other checks in this method later?

Reply:
That what I intended to do, given we have [this trello card](https://trello.com/c/7RIfw0X5/3-05-as-hayao-i-want-to-know-if-my-move-respect-constraints-set-up-by-the-previous-move) next in the pipeline.